File: backend/app/schemas/notification.py
"""Pydantic schemas for in-app notifications"""

import uuid
from datetime import datetime
from typing import Optional
from pydantic import BaseModel, ConfigDict

from app.models.notification import NotificationType


class NotificationResponse(BaseModel):
    model_config = ConfigDict(from_attributes=True)
    id: int
    public_id: uuid.UUID
    type: NotificationType
    title: str
    message: str
    entity_type: Optional[str] = None
    entity_id: Optional[int] = None
    is_read: bool
    read_at: Optional[datetime] = None
    created_at: datetime


class UnreadCountResponse(BaseModel):
    unread_count: int

File: frontend/src/models/notification.ts
export type NotificationType =
	| 'lead_assigned'
	| 'lead_reassigned_away'
	| 'lead_status_changed'
	| 'lead_stale';

/** Matches backend's NotificationResponse exactly. */
export interface Notification {
	id: number;
	public_id: string;
	type: NotificationType;
	title: string;
	message: string;
	entity_type?: string;
	entity_id?: number;
	is_read: boolean;
	read_at?: string;
	created_at: string;
}
